dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheClientGauges.java

```diff
@@ -35,28 +43,60 @@
-    private static final Gauge<Long> leased =
-            () -> collect(pool -> (long) pool.getTotalStats().getLeased(), Long::sum, 0L);
-    private static final Gauge<Long> pending =
-            () -> collect(pool -> (long) pool.getTotalStats().getPending(), Long::sum, 0L);
+    private static final ConcurrentMap<String, List<WeakReference<PoolingHttpClientConnectionManager>>> poolsByName =
+            new ConcurrentHashMap<>();
+    private static final TaggedMetricSet metricSet = new TaggedMetricSet() {
+        @Override
+        public Map<MetricName, Metric> getMetrics() {
+            ImmutableMap.Builder<MetricName, Metric> results = ImmutableMap.builder();
+            forEachMetric(results::put);
+            return results.build();
+        }
+
+        @Override
+        public void forEachMetric(BiConsumer<MetricName, Metric> consumer) {
+            TaggedMetricRegistry registry = new DefaultTaggedMetricRegistry();
+            DialogueClientPoolMetrics poolMetrics = DialogueClientPoolMetrics.of(registry);
+            poolsByName.forEach((channelName, pools) -> {
+                poolMetrics
+                        .size()
+                        .channelName(channelName)
+                        .state("idle")
+                        .build(() -> collect(
+                                pools, pool -> (long) pool.getTotalStats().getAvailable(), Long::sum, 0L));
+                poolMetrics
+                        .size()
+                        .channelName(channelName)
+                        .state("leased")
+                        .build(() -> collect(
+                                pools, pool -> (long) pool.getTotalStats().getLeased(), Long::sum, 0L));
+                poolMetrics
+                        .size()
+                        .channelName(channelName)
+                        .state("pending")
+                        .build(() -> collect(
+                                pools, pool -> (long) pool.getTotalStats().getPending(), Long::sum, 0L));
+            });
+            registry.forEachMetric(consumer);
+        }
+    };
 
-    static void register(PoolingHttpClientConnectionManager connectionPool) {
+    static void register(String channelName, PoolingHttpClientConnectionManager connectionPool) {
+        List<WeakReference<PoolingHttpClientConnectionManager>> pools =
+                poolsByName.computeIfAbsent(channelName, ignored -> new CopyOnWriteArrayList<>());
         pools.add(new WeakReference<>(connectionPool));
         pools.removeIf(item -> item.get() == null);
     }
 
     static void install(TaggedMetricRegistry registry) {
-        DialogueClientPoolMetrics poolMetrics = DialogueClientPoolMetrics.of(registry);
-        poolMetrics.size().state("idle").build(idle);
-        poolMetrics.size().state("leased").build(leased);
-        poolMetrics.size().state("pending").build(pending);
+        registry.addMetrics("set", "apacheClientGauges", metricSet);
```

Comment:
any concerns if I just PR tritium to let us just `addMetrics(TaggedMetricSet)` without having to specify the extra tag names?

Reply:
I think metric sets are identified by a tag, I suppose a concurrent set would be fine if the original is tracked for removal. However it becomes easy to foot-gun two instances of the same type of metric set in a way that causes duplicate keys, which makes getMetrics throw.

Reply:
I think that exact foot-gun (with the current API) was the cause of @jkozlowski's disappearing metrics PDS ticket. I think it's not significantly more dangerous tbh, and I kinda like the ability to compose things without having to add throwaway strings into the mix

Reply:
I don't have a strong opinion either way, defer to dave